test/utils/BaseTest.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import org.scalatest.mockito.MockitoSugar
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.i18n.{I18nSupport, MessagesApi}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.test.UnitSpec
+
+import scala.concurrent.ExecutionContext
+
+trait BaseTest extends UnitSpec with I18nSupport with GuiceOneAppPerSuite with MockitoSugar {
+
+  implicit val ec = ExecutionContext.Implicits.global
```

Comment:
isnt there an ExecutionContext we can inject

Reply:
Do you mean via an import?

Reply:
DI, in the same way we are injecting below

Reply:
```suggestion
  implicit val ec : ExecutionContext = app.injector.instanceOf[ExecutionContext]
```

Reply:
please also give implicit val type signature, more important than the naming for implicits

Reply:
fixed